class.jetpack-signature.php

```diff
@@ -204,6 +204,17 @@ function sign_request( $token = '', $timestamp = 0, $nonce = '', $body_hash = ''
 		);
 
 		$normalized_request_pieces = array_merge( $normalized_request_pieces, $this->normalized_query_parameters( isset( $parsed['query'] ) ? $parsed['query'] : '' ) );
+		$flat_normalized_request_pieces = array();
+		foreach ($normalized_request_pieces as $piece) {
+			if ( is_array( $piece ) ) {
+				foreach ($piece as $subpiece) {
+					$flat_normalized_request_pieces[] = $subpiece;
+				}
```

Comment:
Since this isn't recursive, it will only work up to two levels deep, correct?

Reply:
That's correct 👍

Reply:
Do we want it to be flexible enough for more than that, or YAGNI? :slightly_smiling_face:

Reply:
> Do we want it to be flexible enough for more than that

(Asking because `encode_3986` now works recursively.)

Reply:
Good question, but nah - YAGNI indeed.